### PR TITLE
[CCIP-2937] Fixes Conflicts and Run Issues with CCIP E2E Tests Merge

### PR DESCRIPTION
**Pull request by kalverra**
Merges in fixes for the initial issues found with plain copy-paste of CCIP code into core. There is much more to do, which is tracked by TT-1441. And especially updates to CI needed, tracked by TT-1448.

### CODE REVIEW

File: integration-tests/ccip-tests/actions/ccip_helpers.go [hunk 1]
```diff
@@ -4036,7 +4035,7 @@ func (c *CCIPTestEnv) ConnectToDeployedNodes() error {
 		for _, chainlinkNode := range c.LocalCluster.ClCluster.Nodes {
 			c.nodeMutexes = append(c.nodeMutexes, &sync.Mutex{})
 			c.CLNodes = append(c.CLNodes, &client.ChainlinkK8sClient{
-				ChainlinkClient: chainlinkNode.API.WithRetryCount(3),
```
Comment (asoliman92) [inline integration-tests/ccip-tests/actions/ccip_helpers.go:4039]: What's the reason for removing the retry?

Comment (kalverra) [reply integration-tests/ccip-tests/actions/ccip_helpers.go:4039]: This seems to be an ancient kludge from when our deployments were much more unstable. We haven't had this function call for a long time in the core repo.

**Review by asoliman92 — APPROVED** [review]
LGTM

File: .github/workflows/ccip-offchain-upgrade-tests.yml [hunk 2]
```diff
@@ -1,390 +0,0 @@
-name: CCIP Offchain Upgrade Compatibility Tests
```
Comment (AnieeG) [inline .github/workflows/ccip-offchain-upgrade-tests.yml:1]: Is there any alternative?

**Review by AnieeG — COMMENTED** [review]
We might have to think if we should change test image from` chainlink-ccip-tests `to the usual test image for chainlink

File: .github/workflows/ccip-offchain-upgrade-tests.yml (hunk 2 above)
Comment (kalverra) [reply .github/workflows/ccip-offchain-upgrade-tests.yml:1]: For now it's messy. If we fix it to run okay now, it risks testing the wrong version upgrade path and giving us false negatives. If we leave it broken, it risks people disregarding it as a flake in the future. The plan is to revive it [TT-1448] when we've ironed out the new release process for CCIP in the main repo.

**Review by AnieeG — APPROVED** [review]

**Pull request merged**